content/articles/spam-detection-model-using-scikit-learn/index.md

```diff
@@ -0,0 +1,338 @@
+
+### Table of contents
+
+- [Prerequisites](#prerequisites)
+- [Dataset preparation](#dataset-preparation)
+- [Extracting important columns](#extracting-important-columns)
+- [Feature extraction from text](#feature-extraction-from-text)
+- [Model building](#model-building)
+- [Accuracy score of our model](#accuracy-score-of-our-model)
+- [Model evaluation](#model-evaluation)
+- [Making a single prediction](#Making-a-single-prediction)
+- [Making another prediction](#making-another-prediction)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Prerequisites
+
+A reader should know the following to have a clear understanding of this tutorial.
+
+- Be well equipped with [Python programming](/engineering-education/python-projects-for-beginners/) skills.
+- Understand the concepts of [machine learning.](/engineering-education/house-price-prediction/)
+- Have some knowledge about [natural language processing](/engineering-education/nlp-based-detection-model-using-neattext-and-scikit-learn/).
+- Know how to work with some of the [Scikit-learn](https://scikit-learn.org/stable/) algorithms.
+- Know how to build a machine learning model using [Google Colab notebook](https://research.google.com/).
+
+### Dataset preparation
+
+The dataset used contains a list of comments from popular Youtube channels. We will use a dataset collected from five Youtube channels. We need to prepare this dataset so that it can be ready for use. 
+Data preparation involves correctly formatting our dataset to make it easy for use by the model during training.
+
+First, we need to load these datasets into our machine. Let's import the packages that will load our dataset.
+
+```python
+import pandas as pd
+import numpy as np
+```
+We will use Pandas to read the datasets and Numpy will be used to perform mathematical operations on these datasets. Since we have collected the dataset from five Youtube channels, we will have a total of five datasets. 
+
+To download the five datasets in a ZIP file, click [here](https://drive.google.com/file/d/1rF1qv4onQQ-DH0MRJdVtKnm0pAApDk96/view?usp=sharing). After downloading the ZIP file, extract the individual datasets which we will load onto our machine.
+
+To load the five datasets, use the following code:
+
+```python
+df1 = pd.read_csv("Youtube01-Channel1.csv")
+df2 = pd.read_csv("Youtube02-Channel2.csv")
+df3 = pd.read_csv("Youtube03-Channel3.csv")
+df4 = pd.read_csv("Youtube04-Channel4.csv")
+df5 = pd.read_csv("Youtube05-Channel5.csv")
+```
+Now we have five datasets, we need to concatenate or merge them. The five datasets will be joined together so that we have a single data frame.
+
+#### Datasets concatenation
+
+To join them together, we create a single data frame for the datasets and then apply the `concat` method.
```

Comment:
Please remove the white spaces after the heading. It's unnecessary as it's automatically added during rendering. Perform this process on the whole document.